src/api/controllers/auth.controller.ts

```diff
@@ -0,0 +1,68 @@
+import { Request, Response } from 'express';
+import * as jwt from 'jsonwebtoken';
+import { Service } from 'typedi';
+
+import { Config } from '../../config';
+import { HandleError, LogRequestData } from '../../decorators';
+import { RefreshTokenService, UserService } from '../../services';
+import { HttpCode, HttpMessage } from '../constants';
+
+@Service()
+export class AuthController {
+    constructor(private userService: UserService, private refreshTokenService: RefreshTokenService) {}
+
+    @LogRequestData()
+    @HandleError()
+    async login(req: Request, res: Response) {
+        const { login, password } = req.body;
+        const user = await this.userService.getUserByLogin(login);
+
+        if (!user) {
+            return res.status(HttpCode.NotFound).send(HttpMessage.NotFound);
+        }
+
+        if (password !== user?.password) {
+            return res.status(HttpCode.Forbidden).send(HttpMessage.Forbidden);
+        }
+
+        const accessToken = jwt.sign({ sub: user.id }, Config.auth.secret, { expiresIn: Config.auth.jwtExpiration });
+        const refreshToken = await this.refreshTokenService.createToken(user.id);
+
+        if (!refreshToken) {
+            return res.status(HttpCode.InternalServerError).send(HttpMessage.InternalServerError);
+        }
+
+        return res.status(HttpCode.Ok).json({ accessToken, refreshToken: refreshToken.token });
+    }
+
+    @LogRequestData()
+    @HandleError()
+    async refreshAccessToken(req: Request, res: Response) {
+        const { userId, refreshToken } = req.body;
+        const refreshTokenDTO = await this.refreshTokenService.getTokenByUserId(userId);
```

Comment:
I don's think that we should limit the number of refresh tokens to 1 per user

Reply:
@KonanMentor I have corrected it in the way one user could has several refresh tokens (one to many relations) but only one not expired. In case of login from different devices at the same time each of them will receive the same refresh token in my implementation